build.mill

```diff
@@ -216,3 +218,45 @@ object os extends Module {
     }
   }
 }
+
+trait ApacheAntZipVendor extends Module {
```

Comment:
This can probably be a normal `object` rather than a `trait`.

It also deserves a comment summarizing the reasons:

* Why we chose to include a third party implementation (because the JDK version doesn't support symlinks) 
* Why we chose the Ant implementation over apache-commons (to avoid a dependency on apache-commons-io, and anyway it's where the commons-io implementation came from) 
* Why we are shading it (to ensure it doesn't cause classpath conflicts)
* Why we are building it from source (because we only want a subset of the codebase, not the entire thing)

Reply:
Let's also compile the shaded apache zip stuff in a `os.zip` submodule in `os/zip/src/` separate from the main OS-Lib codebase in `os/src/`. It's generally best to avoid mixing generated and hand-written code.